libbeat/outputs/logstash/config.go

```diff
@@ -11,6 +11,16 @@ type logstashConfig struct {
 	CompressionLevel int                `config:"compression_level"`
 	MaxRetries       int                `config:"max_retries"`
 	TLS              *outputs.TLSConfig `config:"tls"`
+	SOCKSProxy       *socksProxyConfig  `config:"socks_proxy"`
+}
+
+// socksProxyConfig holds the configuration information required to proxy
+// Logstash connections through a SOCKS5 server.
+type socksProxyConfig struct {
+	HostPort  string `config:"hostport"`   // host:port of the SOCKS5 server.
```

Comment:
How about I replace  `HostPort`, `Username`, and `Password` with `ProxyUrl` (`proxy_url` in config)? This is more consistent with how you set the proxy for ES (it's also `proxy_url`).

So you would write:

```
socks_proxy:
  proxy_url: socks5://username:password@mysocksproxy:1234
  remote_dns: true
```